src/main/java/at/hannibal2/skyhanni/features/foraging/CompactSweepDetails.kt

```diff
@@ -0,0 +1,202 @@
+        }
+        if (isInsideSweepDetails) {
+            sweepToughnessLogsPattern.matchMatcher(message) {
+                treeType = group("treeType")
+                toughnessDisplay = group("toughnessDisplay")
+                toughness = group("toughnessAmount").formatDouble()
+                logCountDisplay = group("logsDisplay")
+                logs = group("logsAmount").formatDouble()
+                val fixedToughness = toughness.toString().removeSuffix(".0")
+                sweepPenaltyHoverHistory.add("§6Initial Logs: $logs $treeType Logs §7(§6$fixedToughness toughness§7)")
+                blockedReason = "SWEEP_DETAILS"
+                if (isFinalCalculation(group("isItGreen"))) {
+                    sendCompactedResults()
+                }
+            }
+            penaltyPattern.matchMatcher(message) {
+                if (!addedInitialLogs) {
+                    sweepDetailsChatBreakdown.add("§7, §e$logs logs")
+                    addedInitialLogs = true
+                }
+                logs = group("logsAmount").formatDouble()
+                logCountDisplay = group("logsDisplay")
+                sweepPenaltyHoverHistory.add("§e${group("penaltyReason")}§7: §c-${group("penaltyPercent").formatDouble()}% §7($logs logs)")
+                sweepDetailsChatBreakdown.add("§7(${group("penaltyDisplay")}§7)")
+                proTip = groupOrEmpty("proTip")
+                blockedReason = "SWEEP_DETAILS"
+                if (isFinalCalculation(group("isItGreen"))) {
+                    sendCompactedResults()
+                }
+            }
+        }
+    }
+
+    private fun sendCompactedResults() {
+        sweepPenaltyHoverHistory.add("§6Final Logs: §a$logs §6$treeType Logs")
+        isInsideSweepDetails = false
+
+        val builder = StringBuilder()
+
+        sweepPenaltyHoverHistory.forEach { penalty ->
+            builder.append(penalty)
+            if (penalty != sweepPenaltyHoverHistory.last()) builder.append("\n")
+        }
+        if (proTip.isNotEmpty()) builder.append("\n§6Pro tip: $proTip")
+        val hoverText = builder.toString()
+        val hoverComponent = hoverText.asComponent()
+
+        builder.clear()
+
+        sweepDetailsChatBreakdown.forEach { section ->
+            builder.append(section)
+            if (!section.startsWith("§2Sweep: ") || !addedInitialLogs) builder.append(" ")
+        }
+        builder.append("§7-> §a$logs logs")
+        if (proTip.isNotEmpty()) builder.append("\n  §6Pro tip: $proTip")
+        val chatText = builder.toString()
+        val chatComponent = chatText.asComponent()
+
+        chatComponent.hover = hoverComponent
+        chatComponent.onClick(onClick = {
+            HypixelCommands.treeGifts()
+        })
+
+        ChatUtils.chat(chatComponent)
+        resetSweepDetailsVariables()
+    }
+
+    private fun isFinalCalculation(regexGroup: String): Boolean = regexGroup == "§a"
+
+    private fun resetSweepDetailsVariables() {
```

Comment:
instead of manually reset all those details here, please use a data class that you set to null once

Reply:
opting for data class inherting from abstract `ResettableStorageSet` class, hope you won't mind